wp-job-manager-functions.php

```diff
@@ -98,12 +98,21 @@ function get_job_listings( $args = [] ) {
 		if ( ! empty( $args['search_location'] ) ) {
 			$location_meta_keys = [ 'geolocation_formatted_address', '_job_location', 'geolocation_state_long' ];
 			$location_search    = [ 'relation' => 'OR' ];
-			foreach ( $location_meta_keys as $meta_key ) {
-				$location_search[] = [
-					'key'     => $meta_key,
-					'value'   => $args['search_location'],
-					'compare' => 'like',
-				];
+			$locations          = explode( ';', $args['search_location'] );
+
+			foreach ( $locations as $location ) {
```

Comment:
🛸 why do we need to copy the same code here as in includes/class-wp-job-manager-post-types.php ? (I am 100% ignorant about this codebase, just passing by and curious!)

Reply:
> Looking here, I think it is because the get_job_listings necessarily return the WP_Query AND has some extra actions and hooks for customization, while the logic in includes/class-wp-job-manager-post-types.php uses query_post to change the global object.

This is what I understood as well, looking at it last night. There are things specific to the RSS Feed rendering that results in the difference between the two:

https://github.com/Automattic/WP-Job-Manager/blob/b1184b2836bba567218281c0a13f68694c10dd55/wp-job-manager-functions.php#L101

https://github.com/Automattic/WP-Job-Manager/blob/b1184b2836bba567218281c0a13f68694c10dd55/includes/class-wp-job-manager-post-types.php#L595

Namely, `query_posts` in the one instance.

Reply:
ohhh ok, that makes sense to me :) thank you!